filters/tee/tee.go

```diff
@@ -236,43 +295,71 @@ func (spec *teeSpec) CreateFilter(config []interface{}) (filters.Filter, error)
 		return nil, filters.ErrInvalidFilterParameters
 	}
 
-	if u, err := url.Parse(backend); err == nil {
-		tee.host = u.Host
-		tee.scheme = u.Scheme
-	} else {
+	u, err := url.Parse(backend)
+	if err != nil {
 		return nil, err
 	}
 
-	if len(config) == 1 {
-		tee.typ = asBackend
-		return &tee, nil
+	var client *net.Client
+	teeClients.mu.Lock()
+	if client, ok = teeClients.store[u.Host]; !ok {
+		client = net.NewClient(net.Options{
+			Timeout:                 spec.options.Timeout,
+			TLSHandshakeTimeout:     spec.options.Timeout,
+			ResponseHeaderTimeout:   spec.options.Timeout,
+			CheckRedirect:           checkRedirect,
+			MaxIdleConns:            spec.options.MaxIdleConns,
+			MaxIdleConnsPerHost:     spec.options.MaxIdleConnsPerHost,
+			IdleConnTimeout:         spec.options.IdleConnTimeout,
+			Tracer:                  spec.options.Tracer,
+			OpentracingComponentTag: "skipper",
+			OpentracingSpanName:     spec.Name(),
+		})
+	}
+	teeClients.mu.Unlock()
+
+	tee := tee{
+		client: client,
+		host:   u.Host,
+		scheme: u.Scheme,
 	}
 
-	//modpath
-	if len(config) == 3 {
+	switch len(config) {
+	case 1:
+		tee.typ = asBackend
+		return &tee, nil
+	case 3:
+		// modpath
 		expr, ok := config[1].(string)
 		if !ok {
+			client.Close()
```

Comment:
I think we should not close it after we put it into global lookup.
Maybe it is even fine to leave it there even if we return error

Reply:
yeah that was also a feeling that I have. I think we should follow that as we are 2 having that :)